src/m365/spo/commands/file/file-roleassignment-remove.ts

```diff
@@ -0,0 +1,211 @@
+          return `Specified principalId ${args.options.principalId} is not a number`;
+        }
+
+        if (args.options.fileId && !validation.isValidGuid(args.options.fileId)) {
+          return `${args.options.fileId} is not a valid GUID`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push(
+      ['fileUrl', 'fileId'],
+      ['upn', 'groupName', 'principalId']
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    const removeRoleAssignment: () => Promise<void> = async (): Promise<void> => {
+      try {
+        const fileURL: string = await this.getFileURL(args);
+
+        if (args.options.groupName) {
+          args.options.principalId = await this.GetGroupPrincipalId(args.options);
+        }
+        else if (args.options.upn) {
+          args.options.principalId = await this.GetUserPrincipalId(args.options);
+        }
+
+        const serverRelativeUrl: string = urlUtil.getServerRelativePath(args.options.webUrl, fileURL);
+        const requestOptions: AxiosRequestConfig = {
+          url: `${args.options.webUrl}/_api/web/GetFileByServerRelativeUrl('${serverRelativeUrl}')/ListItemAllFields/roleassignments/removeroleassignment(principalid='${args.options.principalId}')`,
+          method: 'POST',
+          headers: {
+            accept: 'application/json;odata.metadata=none'
+          },
+          responseType: 'json'
+        };
+
+        await request.post(requestOptions);
+      }
+      catch (err: any) {
+        this.handleRejectedODataJsonPromise(err);
+      }
+    };
+
+    if (args.options.confirm) {
+      await removeRoleAssignment();
+    }
+    else {
+      const result = await Cli.prompt<{ continue: boolean }>({
+        type: 'confirm',
+        name: 'continue',
+        default: false,
+        message: `Are you sure you want to remove role assignment from file ${args.options.fileUrl || args.options.fileId} from site ${args.options.webUrl}?`
+      });
+
+      if (result.continue) {
+        await removeRoleAssignment();
+      }
+    }
+  }
+
+  private async getFileURL(args: CommandArgs): Promise<string> {
+    if (args.options.fileUrl) {
+      return args.options.fileUrl;
+    }
+
+    const options: SpoFileGetCommandOptions = {
+      webUrl: args.options.webUrl,
+      id: args.options.fileId,
+      output: 'json',
+      debug: this.debug,
+      verbose: this.verbose
+    };
+
+    const output = await Cli.executeCommandWithOutput(SpoFileGetCommand as Command, { options: { ...options, _: [] } });
+    const getFileOutput = JSON.parse(output.stdout);
+    return getFileOutput.ServerRelativeUrl;
+  }
+
+  private async GetUserPrincipalId(options: Options): Promise<number> {
+    const userGetCommandOptions: SpoUserGetCommandOptions = {
+      webUrl: options.webUrl,
+      email: options.upn,
+      id: undefined,
```

Comment:
Can't we just remove the `id` property since it's `undefined` anyway?

Reply:
Hi @milanholemans 
Since `id` is defined as 'id: string | number | undefined' inside user-get, we need to pass it as `undefined`.